llvm/lib/Target/AIE/AIEInterBlockScheduling.h

```diff
@@ -314,10 +315,17 @@ class InterBlockScheduling {
   /// This represents the resource usage we assume for an unscheduled successor.
   std::optional<int> getBlockedResourceCap(MachineBasicBlock *BB) const;
 
-  /// Return the number of nops that must be inserted before this block.
-  /// This is typical for epilogue blocks, which supply the safety margin of
-  /// their loops.
-  int getNumEntryNops(const BlockState &BS) const;
+  /// Return the number of nops that must be inserted before the epilogue
+  /// of the loop represented by this block.
+  int getSafetyMargin(MachineBasicBlock *Loop,
+                      MachineBasicBlock *Epilogue) const;
```

Comment:
Why change references to pointers?

Reply:
I don't understand 'change'. This is a new interface. I can make them references. It could be that my first version could have a nullptr Epilogue or Loop to overload the loop and normal interblock case.

Reply:
Correct, `change` does not make sense :D

Still, if you expect the pointers to be non-null, I think a reference is better